variables.go

```diff
@@ -137,15 +143,27 @@ func (r *reference) resolveEnv(cfg *Config, opts *options) (string, error) {
 
 func (r *reference) resolve(cfg *Config, opts *options) (value, error) {
 	v, err := r.resolveRef(cfg, opts)
-	if v != nil || err != nil {
+	if v != nil || stopResolving(err) {
 		return v, err
 	}
 
+	previousErr := err
+
 	s, err := r.resolveEnv(cfg, opts)
-	if s == "" || err != nil {
+	if err != nil {
+		// TODO(ph): Not everything is an Error, will do some cleanup in another PR.
```

Comment:
When I was running the test, some things didn't return and Error, I will make a followup PR to fix them.